fuse/server_darwin.go

```diff
@@ -30,3 +32,35 @@ func (ms *Server) systemWrite(req *request, header []byte) Status {
 	}
 	return ToStatus(err)
 }
+
+// for a stream connection we need to have a single reader
+var readLock sync.Mutex
+
+func (ms *Server) systemRead(dest []byte) (int, error) {
+	if osxFuse {
+		return syscall.Read(ms.mountFd, dest)
+	}
+	var n int
+	readLock.Lock()
+	err := handleEINTR(func() error {
+		var err error
+		// read request length
+		n, err = syscall.Read(ms.mountFd, dest[0:4])
+		if n != 4 {
+			return err
+		}
+		l := *(*uint32)(unsafe.Pointer(&dest[0]))
+		offset := 4
+		for offset < int(l) {
+			n, err = syscall.Read(ms.mountFd, dest[offset:l])
```

Comment:
If this returns EINTR, then you will try to read the header again, losing the old header, and everything goes to hell, right?

Reply:
Correct, I need to keep a state.